### ISSUE TITLE
Overall Folder and File Structure should contain a test/ folder

### ISSUE THREAD

**Issue by francisoud** [angular/angular.io#1870]
I'm starting to write tests for a new Angularjs2 1.0.0-rc3 application.
I know that the documentation is a work in progress (#1416), but the folder layout currently explain in the angular [style guide](https://angular.io/styleguide#!#04-06) is seriously **flawed** and it's enforced by [angular-cli](https://github.com/angular/angular-cli).

Currently all tests (*.spec.ts) are stored along the components in src/.
It reminds me of the first days of junit when the norm was to put *Test.java along the other application files.
It's not the norm anymore, a classical java project use [maven project structure](https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html) 
I could find many other examples like [RubyOnRails ](http://guides.rubyonrails.org/getting_started.html#creating-the-blog-application)

In a nutshell, everyone split src/ and test/ because 
* when I write a `app.component.ts` using ->`shared/my.service.ts `
* I always end up adding and configuring more than just a single test file: `app.component.spec.ts`
* I may need a `shared/mock.my.service.ts`
* I may need to configure a `shared/in-memory-data.service.ts` (to mock http request)
* etc.

putting them in different folders makes it obvious that it's only for test purpose.
It also add the possibility to decide (based on simple filenames patterns `test/**/*` or` src/**/*`) what files need to be build or tested with your favorite build tool or testing framework.

FYI
```
angular@2.0.0-rc.3
>ng --version
angular-cli: 1.0.0-beta.8
node: 6.2.1
os: win32 x64

```

**Comment by Foxandxss** [comment]
We don't force anyone to use a certain folder structure. We encourage a team to use a structure that works for them and just be consistent with it.

I am closing this because there is no action we could do as today.

Thanks :)

**Comment by francisoud** [comment]
Just for the record the official style guide says the same:
https://angular.io/docs/ts/latest/guide/testing.html#!#_observable_-test-double

> The hero-detail.component.spec.ts relies on an ActivatedRouteStub to set ActivatedRoute.params values for each test. This is a cross-application, re-usable test helper class. We recommend locating such helpers in a **_testing folder_** sibling to the app folder. This sample keeps ActivatedRouteStub in testing/router-stubs.ts:
